src/data/list/basic.lean

```diff
@@ -2093,6 +2093,14 @@ end
 | b []      l₂ := rfl
 | b (a::l₁) l₂ := by simp only [cons_append, foldr_cons, foldr_append b l₁ l₂]
 
+theorem foldl_fixed (f : α → β → α) {a : α} (hf : ∀ b, f a b = a) : Π l : list β, foldl f a l = a
+| []     := rfl
+| (b::l) := by rw [foldl_cons, hf b, foldl_fixed l]
+
+theorem foldr_fixed (f : α → β → β) {b : β} (hf : ∀ a, f a b = b) : Π l : list α, foldr f b l = b
+| []     := rfl
+| (a::l) := by rw [foldr_cons, foldr_fixed l, hf a]
```

Comment:
Another way of stating these would be
```suggestion
theorem foldl_fixed {a : α} : Π l : list β, foldl (λ a b, a) a l = a
| []     := rfl
| (b::l) := by rw [foldl_cons, foldl_fixed l]

theorem foldr_fixed {b : β} : Π l : list α, foldr (λ a b, b) b l = b
| []     := rfl
| (a::l) := by rw [foldr_cons, foldr_fixed l]
```
which might be better as these would work well as `simp` lemmas

Reply:
It also makes them match the `foldr_const` you have below